scanner/Makefile

```diff
@@ -81,21 +100,21 @@ image-prefix := scanner-v4
 .PHONY: images
 images: scanner-image db-image
 
-scanner-go-build: deps
+image/scanner/bin/scanner: bin/scanner
 	@echo "+ $@"
-	$(BUILD_CMD)
+	$(SILENT)cp $< $@
 
 .PHONY: scanner-go-build-local
 scanner-go-build-local:
 	@echo "+ $@"
-	$(SILENT)$(MAKE) GOOS=${HOST_OS} scanner-go-build
+	$(SILENT)$(MAKE) GOOS=${HOST_OS} image/scanner/bin/scanner
 
 OSSLS_NOTICE_DEP := ossls-notice
 ifdef CI
 	OSSLS_NOTICE_DEP := ossls-notice-no-download
 endif
 .PHONY: scanner-image
-scanner-image: scanner-go-build copy-scripts $(OSSLS_NOTICE_DEP)
+scanner-image: image/scanner/bin/scanner copy-scripts $(OSSLS_NOTICE_DEP)
```

Comment:
why do we need to change scanner-image: scanner-go-build copy-scripts $(OSSLS_NOTICE_DEP) >> image/scanner/bin/scanner , is this going to use the directory "image/scanner/bin/" ?

Reply:
The `scanner-go-build` was initially created to build the scanner binary, which is then used to build the scanner image by [copying it from `image/bin`](https://github.com/stackrox/stackrox/blob/master/scanner/image/scanner/Dockerfile#L16) during the image build. If you look at the original `scanner-go-build`, you'll see it used `BUILD_CMD`, compiling the scanner binary using `-o image/scanner/bin/scanner` on line 41.

The new `build` target is designed to build all commands under `cmd`, including `cmd/scanner`. To make this new `build`, I decided to generate all binaries and save them in `bin/` under the name of their `cmd` directory. As a result, the scanner image, which depends on the path to the scanner binary, needs to copy from that directory, and you will see a target that does that, copying from `bin/scanner` to `image/scanner/bin/scanner`.

Does that answer your question?